### PR TITLE
Leaderboard logic

### PR DESCRIPTION
**Pull request by mzoyinda**
**Description**

This PR fixes # The fetching and resetting of the leaderboard scores at 00:00:00am (midnight)

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: leaderBoard/data2.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[11,0,0,1,2,0,0,3,1,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,1,1,0,0,0,0,1,0,0,1,0,0,0,0,0,0]
```
Comment (kayceeDev) [inline leaderBoard/data2.json]: What data is this?

Comment (mzoyinda) [reply leaderBoard/data2.json]: the solutions data fetched per user, was saved in an array.

Comment (mzoyinda) [reply leaderBoard/data2.json]: but it has now been removed

**Comment by mzoyinda** [comment]
filter feature added and api call reduced to one

File: leaderBoard/index.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+let postPoint = 1;
+let likePoint = 2;
+let solutionPoint = 3;
+
+
+let baseURL = "https://discuss.layer5.io/";
+let config = {
+  headers: {
+    "Api-Key":
```
Comment (kayceeDev) [inline leaderBoard/index.js]: Remove the API keys. Should be private. I think the API query should work without the API key and username

Comment (mzoyinda) [reply leaderBoard/index.js]: yes it should, i would remove them now

**Review by kayceeDev — APPROVED** [review]

**Comment by leecalcote** [comment]
Tossing in the towel?

**Comment by leecalcote** [comment]
@CyrineG how much of a challenge is this? Is this not a database query prettified with html?

**Comment by CyrineG** [comment]
@leecalcote from reading the code the current implementated logic is : make a DB query - > save result in data.js file > call data.js to populate table

There is a function written to refetch that data every day at midnight but it's not called anywhere.

I can think of two options at the top of my head:
1) query the DB first time - > save data in localStorage with timestamp > when displaying table compare the timestamp to see if it's midnight - > if yes, do another query and set new data in localStorage, else use the existing data. 

2) write a sidekiq job for discuss, but that might entail writing a rails plug-in and I think that would overcomplicate things needlessly (might only work if the leaderboard was a rails plug-in too) 

Note: there is another discussion in the other open PR here, about how a discourse theme component (the current implementation) can't create a new page in discuss, so in this case I'm only concidering the fetching and caching of the top 5 contributers data.

**Pull request closed without merging**